pkg/operator/tektonchains.go

```diff
@@ -45,4 +62,139 @@ func EnsureTektonChainsExists(clients chainv1alpha.TektonChainInterface, names u
 		return true, nil
 	})
 	return ks, err
-}
\ No newline at end of file
+}
+
+func VerifySignature(resourceType string) {
+	//Get a signature of taskrun payload
+	resourceUID := cmd.MustSucceed("tkn", resourceType, "describe", "--last", "-o", "jsonpath='{.metadata.uid}'").Stdout()
+	resourceUID = strings.Trim(resourceUID, "'")
+	jsonpath := fmt.Sprintf("jsonpath=\"{.metadata.annotations.chains\\.tekton\\.dev/signature-%s-%s}\"", resourceType, resourceUID)
+	fmt.Println("Waiting 30 seconds")
+	cmd.MustSuccedIncreasedTimeout(time.Second*45, "sleep", "30")
+	signature := cmd.MustSucceed("tkn", resourceType, "describe", "--last", "-o", jsonpath).Stdout()
+	signature = strings.Trim(signature, "\"")
+	//Decode the signature
+	decodedSignature, err := base64.StdEncoding.DecodeString(signature)
+	if err != nil {
+		testsuit.T.Errorf("Error decoding base64")
+	}
+	//Create file with signature
+	file, err := os.Create("sign")
+	if err != nil {
+		testsuit.T.Errorf("Error creating file")
+	}
+	defer file.Close()
+	_, err = file.WriteString(string(decodedSignature))
+	if err != nil {
+		testsuit.T.Errorf("Error writing to file")
+	}
+	//Verify signature with signing-secrets
+	cmd.MustSucceed("cosign", "verify-blob-attestation", "--insecure-ignore-tlog", "--key", public_key_path+"/cosign.pub", "--signature", "sign", "--type", "slsaprovenance", "--check-claims=false", "/dev/null")
+}
+
+func StartKanikoTask() {
+	var tag string = time.Now().Format("010206150405")
+	cmd.MustSucceed("oc", "secrets", "link", "pipeline", "chains-image-registry-credentials", "--for=pull,mount")
+	image := fmt.Sprintf("IMAGE=%s:%s", repo, tag)
+	cmd.MustSucceed("tkn", "task", "start", "--param", image, "--use-param-defaults", "--workspace", "name=source,claimName=chains-pvc", "--workspace", "name=dockerconfig,secret=chains-image-registry-credentials", "kaniko-chains")
+	fmt.Println("Waiting 2 minutes for images to appear in image registry")
+	cmd.MustSuccedIncreasedTimeout(time.Second*130, "sleep", "120")
+}
+
+func GetImageDigestedUrl() (string, string) {
+	// Get Image digest
+	var imageDigest string
+	jsonOutput := cmd.MustSucceed("tkn", "tr", "describe", "--last", "-o", "json").Stdout()
```

Comment:
You can use `-o jsonpath={.status.results}` and it will make following code a little simpler. Even better would be to use go API directly but that would require some more changes.